### PR TITLE
Add the GitHub repo URLs to the Maven metadata

### PR DESCRIPTION
**Pull request by casz**
I doubt dependabot can resolve variables in the URL as it reading XML directly instead of parsing variables.

### CODE REVIEW

**Comment by jglick** [comment]
> I doubt dependabot can resolve variables in the URL

I am not following. Why would Dependabot care about this field?

**Comment by casz** [comment]
See
https://github.com/dependabot/dependabot-core/blob/01e6563639a665ca650fb93c6244ccac0df39fa2/maven/lib/dependabot/maven/metadata_finder.rb#L52

File: pom.xml [hunk 1]
```diff
@@ -37,12 +37,13 @@ THE SOFTWARE.
...
 
   <name>Test harness for Jenkins and plugins</name>
+  <url>https://github.com/jenkinsci/jenkins-test-harness</url>
   <description>Harness used to run functional tests of Jenkins core and plugins.</description>
 
   <scm>
     <connection>scm:git:git://github.com/jenkinsci/${project.artifactId}.git</connection>
     <developerConnection>scm:git:ssh://git@github.com/jenkinsci/${project.artifactId}.git</developerConnection>
-    <url>https://github.com/jenkinsci/${project.artifactId}</url>
+    <url>https://github.com/jenkinsci/jenkins-test-harness</url>
```
Comment (jglick) [inline pom.xml]: Would rather just fix the bug in Dependabot.

Comment (casz) [reply pom.xml]: There are tests that should cover the use case but it's clearly not happening. 🤔

**Pull request merged**